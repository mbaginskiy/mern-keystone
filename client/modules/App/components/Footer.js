import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';

// Import Assets
import bg from '../header-bg.png';

const Component = styled.footer`
  text-align: center;
  padding: 56px 0;
  background: ${props => props.theme.mainLight} url(${bg}) center;
  background-size: cover;
`;

const Text = styled.p`
  margin: 0 0 8px 0;
  font-size: 14px;
  color: #FFF;

  a {
    color: #FFF;
    text-decoration: none;
    font-weight: 700;
  }
`;

const Footer = () => (
  <Component>
    <Text>&copy; 2016 &middot; Hashnode &middot; LinearBytes Inc.</Text>
    <Text>
      <FormattedMessage id="twitterMessage" />
      :
      <a href="https://twitter.com/@mern_io" target="_Blank" rel="noopener noreferrer">
        @mern_io
      </a>
    </Text>
  </Component>
);

export default Footer;
