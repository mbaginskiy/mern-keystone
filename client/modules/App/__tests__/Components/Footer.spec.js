import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Footer from '../../components/Footer';

test('renders the footer properly', () => {
  const wrapper = shallow(<Footer />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('Footer__Text').length).toBe(2);
  expect(wrapper.find('Footer__Text').first().props().children).toBe('© 2016 · Hashnode · LinearBytes Inc.');
});
