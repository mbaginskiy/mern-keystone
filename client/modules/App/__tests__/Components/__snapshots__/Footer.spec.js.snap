// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the footer properly 1`] = `
<Footer__Component>
  <Footer__Text>
    © 2016 · Hashnode · LinearBytes Inc.
  </Footer__Text>
  <Footer__Text>
    <FormattedMessage
      id="twitterMessage"
      values={Object {}}
    />
    :
    <a
      href="https://twitter.com/@mern_io"
      rel="noopener noreferrer"
      target="_Blank"
    >
      @mern_io
    </a>
  </Footer__Text>
</Footer__Component>
`;
